src/java/io/bazel/rulesscala/exe/LauncherFileWriter.java

```diff
@@ -38,10 +41,17 @@ public static void main(String[] args) throws IOException {
             .addJoinedValues("jvm_flags", "\t", jvmFlags)
             .build();
 
-    Path launcher =
-        Paths.get(Runfiles.create().rlocation("bazel_tools/tools/launcher/launcher.exe"));
+    Runfiles runfiles = Runfiles.preload()
+        .withSourceRepository(AutoBazelRepository_LauncherFileWriter.NAME);
+    Path launcher = Paths.get(
+        runfiles.rlocation("bazel_tools/tools/launcher/launcher"));
     Path outPath = Paths.get(location);
 
+    if (! launcher.toFile().exists()) {
+        // Required by Bazel 7 on Windows.
+        launcher = Paths.get(launcher.toString() + ".exe");
```

Comment:
@meteorcloudy This fixes the runfiles problem that broke the Windows builds for Bazel 8 and later in bazelbuild/bazel-central-registry#5490 after bazelbuild/continuous-integration#2350 fixed the symlink problem. The Bazel 7 build passed because it generates `bazel_tools/tools/launcher/launcher.exe`. Bazel 8 and later don't include the `.exe` extension.

Thought this would be good to know in case any other BCR PRs get bitten by this detail.